corehq/apps/case_importer/views.py

```diff
@@ -257,3 +264,73 @@ def excel_commit(request, domain):
     request.session.pop(EXCEL_SESSION_ID, None)
 
     return HttpResponseRedirect(base.ImportCases.get_url(domain))
+
+
+def _get_bulk_case_upload_args_from_request(request, domain):
+    try:
+        upload_file = request.FILES["file"]
+        case_type = request.POST["case_type"]
+        search_field = request.POST['search_field']
+        create_new_cases = request.POST.get('create_new_cases') == 'on'
+        search_column = request.POST['search_column']
+        return upload_file, case_type, search_field, create_new_cases, search_column
+    except Exception:
+        raise ImporterError(
+                "Invalid post request."
+                "Submit the form with field 'file' and the required params")
+
+
+@csrf_exempt
+@require_POST
+@api_auth
+@require_can_edit_data
+def bulk_case_upload_api(request, domain, **kwargs):
+    try:
+        response = _bulk_case_upload_api(request, domain)
+        return response
+    except ImporterError as e:
+        error = get_importer_error_message(e)
+    except SpreadsheetFileExtError:
+        error = "Please upload file with extension .xls or .xlsx"
+        return json_response({'error_msg': _(error)})
+
+
+def _bulk_case_upload_api(request,domain):
+    upload_file, case_type, search_field, create_new_cases, search_column =_get_bulk_case_upload_args_from_request(request, domain)
+
+    case_upload, context = _process_file_and_get_upload(upload_file, request, domain)
+
+    case_upload.check_file()
+
+    with case_upload.get_spreadsheet() as spreadsheet:
+        columns = spreadsheet.get_header_columns()
+        excel_fields = columns
+
+    # hide search column and matching case fields from the update list
+    if search_column in excel_fields:
+        excel_fields.remove(search_column)
+
+    custom_fields = []
+    case_fields = []
+
+    #populate field arrays
```

Comment:
Is this replicated from elsewhere? I am not sure I understand why its doing what its doing, although assume its matching some expected format

Reply:
In the UI driven worklow, this is done in the UI by creating a big table and mapping values in the javascript layer that are then submitted as a form. I'll be honest that I didn't actually go look at that code in detail, I just dumped the output of this object from the other stream and replicated how it built itself